clippy_lints/src/doc/mod.rs

```diff
@@ -502,14 +533,21 @@ fn check_attrs(cx: &LateContext<'_>, valid_idents: &FxHashSet<String>, attrs: &[
     suspicious_doc_comments::check(cx, attrs);
 
     let (fragments, _) = attrs_to_doc_fragments(attrs.iter().map(|attr| (attr, None)), true);
-    let mut doc = String::new();
-    for fragment in &fragments {
-        add_doc_fragment(&mut doc, fragment);
-    }
+    let mut doc = fragments
+        .iter()
+        .fold(String::new(), |mut acc, fragment| {
+            add_doc_fragment(&mut acc, fragment);
+            acc
+        })
+        .trim()
+        .to_string();
```

Comment:
Will need to revert this part, it throws off the spans for other lints because they're reconstructed from the position inside the markdown

Reply:
no the issue here is that I was trimming before popping, should be fixed it.